sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -264,53 +267,90 @@ public Mono<Void> send(List<Message> messageBatch, DeliveryState deliveryState)
-
-                // proton-j doesn't support multiple dataSections to be part of AmqpMessage
-                // here's the alternate approach provided by them: https://github.com/apache/qpid-proton/pull/54
-                final Message batchMessage = Proton.message();
-                batchMessage.setMessageAnnotations(firstMessage.getMessageAnnotations());
-
-                final int maxMessageSizeTemp = maxMessageSize;
-
-                final byte[] bytes = new byte[maxMessageSizeTemp];
-                int encodedSize = batchMessage.encode(bytes, 0, maxMessageSizeTemp);
-                int byteArrayOffset = encodedSize;
-
-                for (final Message amqpMessage : messageBatch) {
-                    final Message messageWrappedByData = Proton.message();
-
-                    int payloadSize = messageSerializer.getSize(amqpMessage);
-                    int allocationSize =
-                        Math.min(payloadSize + MAX_AMQP_HEADER_SIZE_BYTES, maxMessageSizeTemp);
-
-                    byte[] messageBytes = new byte[allocationSize];
-                    int messageSizeBytes = amqpMessage.encode(messageBytes, 0, allocationSize);
-                    messageWrappedByData.setBody(new Data(new Binary(messageBytes, 0, messageSizeBytes)));
-
-                    try {
-                        encodedSize =
-                            messageWrappedByData
-                                .encode(bytes, byteArrayOffset, maxMessageSizeTemp - byteArrayOffset - 1);
-                    } catch (BufferOverflowException exception) {
-                        final String message =
-                            String.format(Locale.US,
-                                "Size of the payload exceeded maximum message size: %s kb",
-                                maxMessageSizeTemp / 1024);
-                        final AmqpException error = new AmqpException(false,
-                            AmqpErrorCondition.LINK_PAYLOAD_SIZE_EXCEEDED, message, exception,
-                            handler.getErrorContext(sender));
-
-                        return Mono.error(error);
+                int totalEncodedSize = 0;
+                final CompositeReadableBuffer buffer = new CompositeReadableBuffer();
+
+                final byte[] envelopBytes = batchEnvelopBytes(messageBatch.get(0), maxMessageSize);
+                if (envelopBytes.length > 0) {
+                    totalEncodedSize += envelopBytes.length;
+                    if (totalEncodedSize > maxMessageSize) {
+                        return batchBufferOverflowError(maxMessageSize);
                     }
+                    buffer.append(envelopBytes);
+                }
 
-                    byteArrayOffset = byteArrayOffset + encodedSize;
+                for (final Message message : messageBatch) {
+                    final byte[] sectionBytes = batchBinaryDataSectionBytes(message, maxMessageSize);
+                    if (sectionBytes.length > 0) {
```

Comment:
Great question. So if the QPid Message object is empty (i.e., none of it properties are set), then _encode_ will return 0; since there is nothing to send for such a message, we cannot add an empty byte[] to CompositeReadableBuffer.

I'll add the logs.